grazel-gradle-plugin/src/main/kotlin/com/grab/grazel/tasks/internal/GenerateBuildifierScriptTask.kt

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Grabtaxi Holdings PTE LTD (GRAB)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.grab.grazel.tasks.internal
+
+import com.grab.grazel.di.qualifiers.RootProject
+import org.gradle.api.DefaultTask
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.register
+import org.gradle.kotlin.dsl.support.serviceOf
+import org.gradle.process.ExecOperations
+
+abstract class GenerateBuildifierScriptTask : DefaultTask() {
+
+    private val execOperations: ExecOperations = project.serviceOf()
+
+    @TaskAction
+    fun action() {
+        execOperations.exec {
+            commandLine = listOf(
+                "bazelisk",
+                "run",
+                "@grab_bazel_common//:buildifier",
+                "--script_path=/tmp/buildifier"
```

Comment:
In addition to above comment, we can declare the `buildifier` script as this task's output and make it an input to format tasks. Gradle will automatically establish dependency and we do not need `dependsOn`.

Ref: https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:task_inputs_outputs